packages/client/src/runtime/core/init/resolveDatasourceUrl.ts

```diff
@@ -29,12 +31,12 @@ export function resolveDatasourceUrl({
 
   // env var is set for use but url is undefined
   if (datasourceUrl?.fromEnvVar !== undefined && resolvedUrl === undefined) {
-    if (TARGET_ENGINE_TYPE === 'edge') {
+    if (TARGET_ENGINE_TYPE === 'edge' && detectRuntime() === 'workerd') {
```

Comment:
Minor nit, non-blocking: let's define a `const jsRuntime = detectRuntime()` block above, and use that here in the conditional.
Realistically, we're going to have custom logic in here for other runtimes as well, and this would reduce the future `git diff`s.